Libraries/LibWeb/CSS/Parser/ValueParsing.cpp

```diff
@@ -383,8 +372,20 @@ RefPtr<CSSStyleValue> Parser::parse_family_name_value(TokenStream<ComponentValue
     if (parts.is_empty())
         return nullptr;
 
+    if (parts.size() == 1) {
+        // <generic-family> is a separate type from <family-name>, and so isn't allowed here.
+        auto maybe_keyword = keyword_from_string(parts.first());
+        if (maybe_keyword.has_value() && keyword_to_generic_font_family(maybe_keyword.value()).has_value()) {
+            return nullptr;
+        }
+    }
+
+    auto complete_name = MUST(String::join(' ', parts));
+    if (is_css_wide_keyword(complete_name) || complete_name.equals_ignoring_ascii_case("default"sv)) {
+        return nullptr;
+    }
```

Comment:
I think you could do both of these checks inside the `parts.size() == 1` block, instead of doing a string join first: they both require that the input is only a single part.

There might also be a version of `is_css_wide_keyword()` somewhere that takes a `Keyword`.